src/providers/wms/qgswmscapabilities.cpp

```diff
@@ -138,6 +138,27 @@ bool QgsWmsSettings::parseUri( const QString &uriString )
   mActiveSubStyles = uri.params( QStringLiteral( "styles" ) );
   QgsDebugMsgLevel( "Entering: layers:" + mActiveSubLayers.join( ", " ) + ", styles:" + mActiveSubStyles.join( ", " ), 2 );
 
+  //opacities
+  if ( uri.hasParam( QStringLiteral( "opacities" ) ) )
+  {
+    mOpacities.clear();
+    QStringList opacities = uri.params( QStringLiteral( "opacities" ) );
+    QStringList::const_iterator oIt = opacities.constBegin();
+    for ( ; oIt != opacities.constEnd(); ++oIt )
+    {
+      bool ok = false;
+      oIt->toInt( &ok );
+      if ( ok )
+      {
+        mOpacities.append( *oIt );
+      }
+      else
+      {
+        mOpacities.append( QStringLiteral( "255" ) );
+      }
+    }
+  }
```

Comment:
Can we this be switched to a range-based for loop for readability?

```suggestion
    const QStringList opacities = uri.params( QStringLiteral( "opacities" ) );
    for ( const QString &opacity : opacities )
    {
      bool ok = false;
      opacity.toInt( &ok );
      if ( ok )
      {
        mOpacities.append( opacity );
      }
      else
      {
        mOpacities.append( QStringLiteral( "255" ) );
      }
    }
  }
```

Reply:
I can do that if you want, no problem.
My personal opinion: const-iterators are still the best. No hidden pitfalls and they are readable as well

Reply:
I share your opinion on the hidden pitfalls, not so much on the readability though.